civis/imports/socrata_helpers.py

```diff
@@ -21,35 +23,88 @@ def _parse_metadata(metadata: dict, paths: dict):
         out[name] = reduce(
             lambda d, key: d.get(key, None) if isinstance(d, dict) else None,
             path.split("."),
-            metadata,
+            metadata
         )
     return out
 
 
+def _store_and_attach_metadata(
```

Comment:
What is the purpose of this function? What are the raw and cleaned metadata, and who is looking at them?

Reply:
Raw metadata contains description, author, columns, column type, min max vals etc. We are using raw metadata (in the new import) to map Socrata datatypes to PostGIS/Redshift datatypes. 

Clean metadata pretty much pulls the description out of metadata and attaches it to the table in platfrom as a description.

Reply:
> 
> Raw metadata contains description, author, columns, column type, min max vals etc. We are using raw metadata (in the new import) to map Socrata datatypes to PostGIS/Redshift datatypes.

Interesting -- where is the code doing this? How does uploading a metadata file to s3 help accomplish getting the datatypes right in the dbs? This is all still pretty opaque to me, my instinct would be to create the create the table with the appropriate data types in this same script, using standard SQL or an ORM.

> 
> Clean metadata pulls the description out of raw metadata and attaches it to the table in platfrom as a description.

:+1:

Reply:
The new code that was just submitted has better documentation and should clear things up. In the meeting today I will go over, at a high level, the code.

Reply:
![image](https://user-images.githubusercontent.com/37056583/86815746-019a6100-c051-11ea-84c8-049ced5d3c32.png)
_store_and_attach_metadata allows us to attach metadata as a JSON (which is stored in S3) to the output of the run and also attaches the Socrata description and author to the table that the data is being stored in. The datatype mapping is not occurring in  _store_and_attach_metadata. It occurs in create_col_type_dict which you can find in the latest commits.